mcs/tools/mono-symbolicate/LocationProvider.cs

```diff
@@ -144,7 +182,151 @@ static bool CompareParameters (Collection<ParameterDefinition> candidate, string
 
 			builder.Append (")");
 
-			return builder.ToString () == expected;
+			if (builder.ToString () == RemoveGenerics (expected, '[', ']'))
+				return true;
+
+			//now try the compact runtime format.
+
+			builder.Clear ();
+
+			builder.Append ("(");
+
+			for (int i = 0; i < candidate.Count; i++) {
+				var parameter = candidate [i];
+				if (i > 0)
+					builder.Append (",");
+
+				if (parameter.ParameterType.IsSentinel)
+					builder.Append ("...,");
+
+				var pt = parameter.ParameterType;
+
+				RuntimeFormatElementType (pt, builder);
+			}
+
+			builder.Append (")");
+
+			if (builder.ToString () == RemoveGenerics (expected, '<', '>'))
+				return true;
+			return false;
+
+		}
+
+		void RuntimeFormatElementType (TypeReference tr, StringBuilder builder)
+		{
+			var ts = tr as TypeSpecification;
+			if (ts != null) {
+				if (ts.IsByReference) {
+					RuntimeFormatElementType (ts.ElementType, builder);
+					builder.Append ("&");
+					return;
+				}
+			}
+
+			switch (tr.MetadataType) {
+			case MetadataType.Void:
+				builder.Append ("void");
+				break;
+			case MetadataType.Boolean:
+				builder.Append ("bool");
+				break;
+			case MetadataType.Char:
+				builder.Append ("char");
+				break;
+			case MetadataType.SByte:
+				builder.Append ("sbyte");
+				break;
+			case MetadataType.Byte:
+				builder.Append ("byte");
+				break;
+			case MetadataType.Int16:
+				builder.Append ("int16");
```

Comment:
No, that's what the runtime outputs.

I should have mentioned that the function is a copy of this one:
https://github.com/mono/mono/blob/master/mono/metadata/debug-helpers.c#L134